test/token/erc20/preset/ImmutableERC20MinterBurnerPermit.t.sol

```diff
@@ -0,0 +1,169 @@
+        vm.expectRevert(abi.encodeWithSelector(IImmutableERC20Errors.RenounceOwnershipNotAllowed.selector));
+        erc20.renounceRole(adminRole, admin);
+    }
+
+    function testRenounceAdmin() public {
+        address secondAdmin = makeAddr("secondAdmin");
+        vm.startPrank(admin);
+        erc20.grantRole(erc20.DEFAULT_ADMIN_ROLE(), secondAdmin);
+        assertTrue(erc20.hasRole(erc20.DEFAULT_ADMIN_ROLE(), secondAdmin));
+        vm.stopPrank();
+
+        vm.startPrank(admin);
+        erc20.renounceRole(erc20.DEFAULT_ADMIN_ROLE(), admin);
+        assertFalse(erc20.hasRole(erc20.DEFAULT_ADMIN_ROLE(), admin));
+        vm.stopPrank();
+    }
+
+    function testRenounceHubOwner() public {
+        address secondHubOwner = makeAddr("secondHubOwner");
+        vm.startPrank(admin);
+        erc20.grantRole(erc20.HUB_OWNER_ROLE(), secondHubOwner);
+        assertTrue(erc20.hasRole(erc20.HUB_OWNER_ROLE(), secondHubOwner));
+        vm.stopPrank();
+
+        vm.startPrank(hubOwner);
+        erc20.renounceRole(erc20.HUB_OWNER_ROLE(), hubOwner);
+        assertFalse(erc20.hasRole(erc20.HUB_OWNER_ROLE(), hubOwner));
+        vm.stopPrank();
+    }
+
+    function testOnlyMinterCanMint() public {
+        address to = makeAddr("to");
+        uint256 amount = 100;
+        vm.prank(hubOwner);
+        vm.expectRevert("AccessControl: account 0xa268ae5516b47694c3f15805a560258dbcdefd08 is missing role 0x4d494e5445525f524f4c45000000000000000000000000000000000000000000");
+        erc20.mint(to, amount);
+    }
+
+    function testMint() public {
+        address to = makeAddr("to");
+        uint256 amount = 100;
+        vm.prank(minter);
+        erc20.mint(to, amount);
+        assertEq(erc20.balanceOf(to), amount);
+    }
+
+    function testBurn() public {
+        uint256 amount = 100;
+        vm.prank(minter);
+        erc20.mint(tokenReceiver, amount);
+        assertEq(erc20.balanceOf(tokenReceiver), 100);
+        vm.prank(tokenReceiver);
+        erc20.burn(amount);
+        assertEq(erc20.balanceOf(tokenReceiver), 0);
+    }
+
+    function testCanOnlyMintUpToMaxSupply() public {
+        address to = makeAddr("to");
+        uint256 amount = 1000;
+        vm.startPrank(minter);
+        erc20.mint(to, amount);
+        assertEq(erc20.balanceOf(to), amount);
+        vm.expectRevert("ERC20Capped: cap exceeded");
+        erc20.mint(to, 1);
+        vm.stopPrank();
+    }
+
+    function testBurnFrom() public {
+        uint256 amount = 100;
+        address operator = makeAddr("operator");
+        vm.prank(minter);
+        erc20.mint(tokenReceiver, amount);
+        assertEq(erc20.balanceOf(tokenReceiver), 100);
+        vm.prank(tokenReceiver);
+        erc20.increaseAllowance(operator, amount);
+        vm.prank(operator);
+        erc20.burnFrom(tokenReceiver, amount);
+        assertEq(erc20.balanceOf(tokenReceiver), 0);
+    }
+
+    function testPermit() public {
+        uint256 ownerPrivateKey = 1;
```

Comment:
makeAddrAndKey() is a nice way to get the address and privatekey in a single line.

`(address alice, uint256 key) = makeAddrAndKey("alice");`